DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -414,6 +428,84 @@ public Task ModifyPositionAsync(int position, string reason = null)
             return this.Discord.ApiClient.ModifyGuildChannelPositionAsync(this.Guild.Id, pmds, reason);
         }
 
+        /// <summary>
+        /// Updates the channel parent, moving the channel to the bottom of the new category.
+        /// </summary>
+        /// <param name="parent_id">New parent ID. Will move out of parent if null.</param>
+        /// <param name="lock_permissions">Sync permissions with parent. Defaults to null.</param>
+        /// <param name="reason">Reason for audit logs.</param>
+        /// <returns></returns>
+        /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageChannels"/> permission.</exception>
+        /// <exception cref="Exceptions.NotFoundException">Thrown when the channel does not exist.</exception>
+        /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public Task ModifyParentAsync(ulong? parent_id = null, bool? lock_permissions = null, string reason = null)
+        {
+            if (this.Guild == null)
+                throw new InvalidOperationException("Cannot modify parent of non-guild channels.");
+            if (this.IsCategory)
+                throw new InvalidOperationException("Cannot modify parent of category channels.");
+
+            var position = this.Guild._channels.Values.Where(xc => xc.Type == this.Type && xc.ParentId == parent_id).Select(xc => xc.Position).DefaultIfEmpty(-1).Max() + 1;
+            var chns = this.Guild._channels.Values.Where(xc => xc.Type == this.Type).OrderBy(xc => xc.Position).ToArray();
```

Comment:
Please use some line breaks, I have no idea what the fuck's going on

Reply:
I'll include line breaks and comments so the logic is easier to follow for the first linq as I wrote that.

> var position = this.Guild._channels.Values.Where(xc => xc.Type == this.Type && xc.ParentId == parent_id).Select(xc => xc.Position).DefaultIfEmpty(-1).Max() + 1;

What this linq does is get the position of the last channel of the same type in the new parent plus one and that will be the new position of the channel being moved. This sets the channel at the bottom of the new cat. If it's empty the position defaults to 0.

It's not perfect, as a news channel being moved to category without any news channels will be placed at the top. But I can rework this later.

> var chns = this.Guild._channels.Values.Where(xc => xc.Type == this.Type).OrderBy(xc => xc.Position).ToArray();

This linq gets the position of all of the channels of the same type within the guild. It's the same as in the `ModifyPositionAsync()` method above it if you want to see an other use of it.

Reply:
Inserted more
Done